src/util/expr.cpp

```diff
@@ -116,48 +146,71 @@ void exprt::make_not()
   swap(new_expr);
 }
 
+/// Return whether the expression is a constant.
+/// \return True if is a constant, false otherwise
 bool exprt::is_constant() const
 {
   return id()==ID_constant;
 }
 
+/// Return whether the expression is a constant representing `true`.
+/// \return True if is a boolean constant representing `true`, false otherwise.
 bool exprt::is_true() const
 {
   return is_constant() &&
          type().id()==ID_bool &&
          get(ID_value)!=ID_false;
 }
 
+/// Return whether the expression is a constant representing `false`.
+/// \return True if is a boolean constant representing `false`, false otherwise.
 bool exprt::is_false() const
 {
   return is_constant() &&
          type().id()==ID_bool &&
          get(ID_value)==ID_false;
 }
 
+/// Replace the expression by a boolean expression representing \p value.
+/// \param value: the boolean value to give to the expression
+/// \deprecated use constructors instead
 void exprt::make_bool(bool value)
 {
   *this=exprt(ID_constant, typet(ID_bool));
   set(ID_value, value?ID_true:ID_false);
 }
 
+/// Replace the expression by a boolean expression representing true.
+/// \deprecated use constructors instead
 void exprt::make_true()
 {
   *this=exprt(ID_constant, typet(ID_bool));
   set(ID_value, ID_true);
 }
 
+/// Replace the expression by a boolean expression representing false.
+/// \deprecated use constructors instead
 void exprt::make_false()
 {
   *this=exprt(ID_constant, typet(ID_bool));
   set(ID_value, ID_false);
 }
 
+/// Return whether the expression represents a boolean.
+/// \return True if is a boolean, false otherwise.
 bool exprt::is_boolean() const
 {
   return type().id()==ID_bool;
 }
 
+/// Return whether the expression is a constant representing 0.
+/// Will consider the following types: ID_integer, ID_natural, ID_rational,
+/// ID_unsignedbv, ID_signedbv, ID_c_bool, ID_c_bit_field, ID_fixedbv,
+/// ID_floatbv, ID_pointer.<br>
+/// For ID_pointer, returns true iff the value is a zero string or a null
+/// pointer.
+/// For all other types, return false.
```

Comment:
:pick: "other" sounds a bit like "for everything other than ID_pointer" after the last sentence, when it should be "for everything not in the above list".

Reply:
Done.